ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -330,6 +334,18 @@ void setVersion(short version) {
     return resetKey;
   }
 
+  /**
+   * @return life version of reset key.
+   */
+  short getResetKeyLifeVersion() throws StoreException {
+    if (resetKey == null || resetKeyLifeVersion != UNINITIALIZED_RESET_KEY_VERSION) {
+      return resetKeyLifeVersion;
+    }
+    NavigableSet<IndexValue> indexValues = find(resetKey.getFirst());
+    // reset key should be present in current index segment, hence indexValues is guaranteed to be non-empty
+    return indexValues.first().getLifeVersion();
+  }
```

Comment:
With a new class, we only need to have one method to return the resetKey info.

Reply:
I am leaning towards to return them separately (3 methods). It has some additional logic in those methods to 1. check whether resetKeyInfo == null, 2. find the correct life version of reset key if not set in V1~V3 index. Thus `ResetKeyInfo` is pure internal data structure, caller doesn't have to know it.